server/src/main/java/com/epam/aidial/core/server/security/IdentityProvider.java

```diff
@@ -182,7 +182,8 @@ public static DecodedJWT decodeJwtToken(String encodedToken) {
     }
 
     private Future<JwkResult> getJwk(String kid) {
-        return cache.computeIfAbsent(kid, key -> vertx.executeBlocking(() -> {
+        Promise<JwkResult> promise = Promise.promise();
+        cache.computeIfAbsent(kid, key -> vertx.executeBlocking(() -> {
```

Comment:
Please add a comment why we don’t reuse the cached result.

Reply:
added